### DISCUSSION TITLE
Redirect input/output on windows

### DISCUSSION THREAD

**Discussion by lemmuh** [mruby/mruby#6254] [Q&A]
I want to redirect the input/output to/from a running app by:

```
IO.popen "cmd.exe /C app.exe", "rw" do |io|
  p io.closed?
  ...
end

```
In mode="rw" there is an ArgumentErrror exception

```
illegal access mode rw (ArgumentError)
```
For mode="w" the io is closed immediately (io.closed? is true).

When the mode is just "r", I can read from the io and the output of the app is available inside the ruby script,
but writing data to the app is forbidden.

How can I get the mode="rw" to work properly ?

**Comment by matz** [comment]
There's no such mode "rw" (CRuby raises error too), use "r+" instead.
